pkg/controller/backup_finalizer_controller.go

```diff
@@ -165,6 +168,7 @@ func (r *backupFinalizerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, errors.WithStack(err)
 		}
 	}
+	defer r.backupTracker.Delete(backup.Namespace, backup.Name)
```

Comment:
This is similar to the above comment.
If the finalizer reconcile error out before here at line 171, the backupTracker will not be deleted, as a result, the backup will never be deletable.

Reply:
In this case, that is deliberate, as the backup won't be in a terminal state if the  func errors out here. However, I'll do the same here as above -- defer statement early, but check for terminal states before removing.